tests/unit/relations/test_db.py

```diff
@@ -0,0 +1,390 @@
+
+        _add_user.assert_called_with(
+            user, password, admin=False, cfg=_read_cfg.return_value, render_cfg=False
+        )
+        _render_cfg.assert_called_with(_read_cfg.return_value, reload_pgbouncer=True)
+
+    @patch("charm.PgBouncerK8sCharm.backend_relation", new_callable=PropertyMock)
+    @patch("charm.PgBouncerK8sCharm.read_pgb_config", return_value=PgbConfig(DEFAULT_CONFIG))
+    @patch("relations.db.DbProvides.get_external_units", return_value=[MagicMock()])
+    @patch("relations.db.DbProvides.get_allowed_units", return_value="test_allowed_unit")
+    @patch("relations.db.DbProvides.get_allowed_subnets", return_value="test_allowed_subnet")
+    @patch("relations.db.DbProvides._get_standbys", return_value="test-postgres-standbys")
+    @patch("relations.db.DbProvides._get_state", return_value="test-state")
+    @patch("charm.PgBouncerK8sCharm.add_user")
+    @patch("charm.PgBouncerK8sCharm._render_pgb_config")
+    def test_admin_user_generated_with_correct_admin_permissions(
+        self,
+        _render_cfg,
+        _add_user,
+        _state,
+        _standbys,
+        _allowed_subnets,
+        _allowed_units,
+        _external_units,
+        _read_cfg,
+        _backend,
+    ):
+        self.harness.set_leader(True)
+        _read_cfg.return_value["databases"]["pg_master"] = {
+            "host": "test-host",
+            "port": "test-port",
+        }
+
+        mock_event = MagicMock()
+        relation_data = mock_event.relation.data = {}
+        database = "test_db"
+        user = "test_user"
+        password = "test_pw"
+
+        relation_data[self.db_relation.charm.unit] = {}
+        relation_data[self.charm.app] = {
+            "database": database,
+            "user": user,
+            "password": password,
+        }
+        external_unit = _external_units.return_value[0]
+        relation_data[external_unit] = {}
+        external_unit.app.name = "external_test_unit"
+
+        self.db_admin_relation._on_relation_changed(mock_event)
+
+        _add_user.assert_called_with(
+            user, password=password, admin=True, cfg=_read_cfg.return_value, render_cfg=False
+        )
+
+    @patch("charm.PgBouncerK8sCharm.backend_relation", new_callable=PropertyMock)
+    def test_get_standbys(self, backend_relation):
+        backend_data = {"read-only-endpoints": "host1:port1,host2:port2"}
+        self.charm.backend_relation.data = {self.charm.backend_relation.app: backend_data}
+
+        cfg = PgbConfig(DEFAULT_CONFIG)
+        app = "app_name"
+        cfg_entry = "db_app"
+        db_name = "dbname"
+        user = "user"
+        pw = "pw"
+
+        standbys = self.db_relation._get_standbys(cfg, app, cfg_entry, db_name, user, pw)
+        standby_list = standbys.split(", ")
+
+        assert len(standby_list) == 2
+
+        assert cfg["databases"][f"{cfg_entry}_standby"] == {
+            "host": "host2",
+            "dbname": db_name,
+            "port": "port2",
+        }
```

Comment:
Why just checking the second endpoint? Shouldn't it be the `{cfg_entry}_standby_1` key?

Reply:
the db relations only expect (and use) one relation, named `f"{database}_standby" - this has been updated to match the new version.